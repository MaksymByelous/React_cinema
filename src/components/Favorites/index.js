import React, { Component } from 'react';
import Styles from './styles.scss';
import AppStyles from '../../containers/App/styles.scss';
import { array } from 'prop-types';


export default class CinemaList extends Component {
    static propTypes = {
        favorites: array.isRequired
    }

    render () {
        const { favorites } = this.props;
        const favoritesList = favorites.map(
            (data, index) =>
                <li key = { index }> { data.title } </li>
        );

        return (
            <div className = { Styles.favorites }>
                <div className = { AppStyles.content }>
                    <h3>My favorites</h3>
                    <ul>
                        { favoritesList }
                    </ul>
                </div>
            </div>
        );
    }
}
