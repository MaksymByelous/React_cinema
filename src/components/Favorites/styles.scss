.favorites {
    padding: 5px;
    background: gray;
    color: white;
}