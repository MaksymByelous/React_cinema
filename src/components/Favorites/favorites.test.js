import React from 'react';
import dom from 'react-test-renderer';
import Favorites from './';

const renderTree = dom.create(
    <Favorites favorites = { ['mike', 'jong'] } />
).toJSON();

test(
    'favorites should correspond to snapshot', () => {
        expect(renderTree).toMatchSnapshot();
    }
);
