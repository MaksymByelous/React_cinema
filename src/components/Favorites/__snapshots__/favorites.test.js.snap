// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorites should correspond to snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <h3>
      My favorites
    </h3>
    <ul>
      <li>
         
         
      </li>
      <li>
         
         
      </li>
    </ul>
  </div>
</div>
`;
